### ISSUE TITLE
Explain Cache Requirements for New Videos

### ISSUE THREAD

**Issue by Jimmbo** [JohnCoates/Aerial#1047]
Using (hallelujah) v2.0....

- [ ] Mac model: Mac Mini (2018)
- [ ] macOS version: 1.15.6
- [ ] Monitor setup: HP Z27

I tried to load the Magic Window videos, and got a pop-up that my cache is full. Cache is currently 5gb. I tried raising it to 10gb and tried again, and was told, again, that my cache was full. 

I guess I could keep guessing, but it would be more efficient if Aerial could explain, on that "Cache Full" pop-up, how much more is required for the videos I'm trying to download...and, perhaps, brief instructions for newbies about where to go to make that adjustment.

**Comment by glouel** [comment]
@Jimmbo so it should still start. What the current size of your cache ? Cause if you had a large cache before, Aerial will not trim it right away, only in a week (if you picked the weekly rotation). 

You can see the current cache size here : 

![Capture d’écran 2020-08-20 à 01 37 53](https://user-images.githubusercontent.com/37544189/90700381-fa888480-e285-11ea-9d15-bf1df5191728.jpg)

This may not be super intuitive for users upgrading like you, I may need to rework this a bit !

**Comment by Jimmbo** [comment]
I answered that above (did I use the wrong terminology?). I currently have cache limited to 5gb. I tried increasing to 10gb, but it still told me cache was full.

**Comment by glouel** [comment]
I mean, not how much is your limit, check what I circled, it tells you your current cache size. You probably had a very large cache from before, and Aerial didn't delete your videos right away.

**Comment by dscos** [comment]
Does 2.0 not allow videos to be streamed anymore?

**Comment by Jimmbo** [comment]
Yes, that's the settings pane I've been referring to. I'm not sure what you thought I meant. 

It was at 5GB, and I increased to 10GB.

**Comment by glouel** [comment]
That’s the limit you set you are talking about. Check the part I circled, I’m not talking about that, at the right of where you set the limit, it says (currently x GB), that’s the current real size of your cache.

**Comment by Jimmbo** [comment]
Sorry. My English is not as good as yours, obviously.

Currently at 31.39GB.

**Comment by glouel** [comment]
@Jimmbo no worries, I was a bit tired too yesterday after all the work on release so I probably wasn't super clear.

So right now if you want new videos to download, you need to move the slider above 32 GB so it will pick new videos. I'll make a fix in next version so it asks you instead if you want to trim your cache to make some room, and make that UI a bit more clear.

@dscos no sorry, it was not really supported previously, there are many issues with streaming with Apple servers (technical and ethical) so I'm no longer allowing that. With a small size cache you should still get a decent solution.

**Comment by Jimmbo** [comment]
Just to organize thoughts....

Any time a user chooses to download videos, creating cache problems, one of these responses is necessary, in descending order of preference:

1. "These videos will take up space on your hard drive, so Aerial will need to increase its cache (disk space allocation) to XX. Is that ok?"

or

2.  "These videos will take up space on your hard drive, so you will need to increase Aerial's cache (disk space allocation) to XX. Press "ok" to open your Desktop&Screensaver control panel, then choose "Aerial/Screen Saver Options/Cache" and move the slider to XX or higher."

or

3. "These videos will take up space on your hard drive, so you will need to increase Aerial's cache (disk space allocation) to XX. Please go to "System Preferences/Desktop&Screensaver/Aerial/Screen Saver Options/Cache and move the slider to XX or higher."


Simply complaining about cache is insufficient.

**Comment by glouel** [comment]
Thanks a lot for the recap, I'm currently fixing a few things including this, I haven't replied to all your posts but I see them and will when I get a sec but agreed on this.

**Comment by Jimmbo** [comment]
Don't worry, there's no rush. But maybe for v3, have me beta test :)

**Comment by glouel** [comment]
You are very welcome ;) Make sure you set the companion app to give you beta updates so you don't miss it next time !

**Comment by glouel** [comment]
> 1. "These videos will take up space on your hard drive, so you will need to increase Aerial's cache (disk space allocation) to XX. **Please go to "System Preferences/Desktop&Screensaver/Aerial/Screen Saver Options/Cache and move the slider to XX or higher."**

To be clear, are you saying that you got the warning when the screen saver settings were closed ? I'm confused here, you should only get that warning when you do an action manually in the screen saver settings.

**Comment by Jimmbo** [comment]
I'm still on the same issue I raised in opening this thread:

> 
> I tried to load the Magic Window videos, and got a pop-up that my cache is full. Cache is currently 5gb. I tried raising it to 10gb and tried again, and was told, again, that my cache was full.
> 
> I guess I could keep guessing, but it would be more efficient if Aerial could explain, on that "Cache Full" pop-up, how much more is required for the videos I'm trying to download...and, perhaps, brief instructions for newbies about where to go to make that adjustment.


Your solution was to send me to adjust my cache. I think it should either happen transparently, or assisted, or at least better instructed.

**Comment by glouel** [comment]
I understand your suggestion. Some of it is not possible, for example I don't know how large the files are before starting the download. But I'll figure something. 

My question was about whether you saw the alert message when the screensaver was running full screen, but I think you didn't.

**Comment by Jimmbo** [comment]
Correct, that didn't happen.

If you can't make it transparent, at least offer explanation. Explanation is the lowest form of programming!  :)

I'm happy to help write stuff. I love the project, glad to help.

**Comment by Jimmbo** [comment]
Also,

>  I don't know how large the files are before starting the download. 

This is eventually going to be a headache for you, I think. 

For example, Magic Window's "Complete Aerial Expansion Bundle" consists of 150 4K videos (20-60 secs in length). Magic Window is not advertising the size of that bundle (it must be enormous), and it sounds like Aerial won't detect it, either. 

If Magic Window's stuff catches on, or if users start adding lots of other video, cache management and file size will require more thorough and clever control. I don't have suggestions, I just see a problem. I predict that Aerial will need better visibility/management of disk space, file size, and cache size, if only as a consequence to the (wonderful) Magic Window association.

If I'm misunderstanding the tech, just ignore me.

**Comment by glouel** [comment]
It's a bit more complex than that, short answer, no those packs are not enormous at all, the videos are well compressed and high quality. Can't go into details right now sorry, need to push my build.

**Comment by Jimmbo** [comment]
Ordering now then. Thanks.

**Comment by glouel** [comment]
Great, I pushed my build to notarization, so to clarify things a bit, the problem you experienced is because you had a large cache, then enabled the cache limit, and I didn't purge your cache of the excess of videos at that time. So despite having a limit set to 10 GB, you already had, from a previous version of Aerial, 31 GB of videos. Does that part make sense so far ? 

What I should have done, is that after you play with the slider, I purge your cache of the excess of videos. I'll implement this later on. 

A second thing I'll try, is to increase the cache after the downloads are done if you pick download anyway. 

For now, 2.0.1, I clarified a bit the error message but I'll revisit this : 
<img width="372" alt="Capture d’écran 2020-08-21 à 00 06 33" src="https://user-images.githubusercontent.com/37544189/90830940-f2dae580-e342-11ea-8a41-11a5bd2335b2.png">

About the packs, they are stored in a separate folder and not in the cache directory, and since you paid for the content, I won't delete it independently of what you set for the cache size. I will however improve at some point the cache UI to more clearly show how much space is currently used, and by what.

**Comment by Jimmbo** [comment]
All sounds good to me.

**Comment by czo** [comment]
> Does 2.0 not allow videos to be streamed anymore?

I am still used the pre 2.0 version because of the forced cache usage things (i really don't want to cache theese videos, because i am almos always online and if i am not, the black screen are perfect for me), but now i made a new fork of the last 1.9.x version. It now contains the tvOS15 videos and i have a lot's of plan to remove lot's of things because i really need nothing more, than a simple screensaver that works very similary to the Apple's own version in tvOS.

**Comment by czo** [comment]
> Does 2.0 not allow videos to be streamed anymore?

@dscos: I am still used the pre 2.0 version because of the forced cache usage things (i really don't want to cache theese videos, because i am almos always online and if i am not, the black screen are perfect for me), but now i made a new fork of the last 1.9.x version. It now contains the tvOS15 videos and i have a lot's of plan to remove lot's of things because i really need nothing more, than a simple screensaver that works very similary to the Apple's own version in tvOS.
